packages/flutter/test/cupertino/segmented_control_test.dart

```diff
@@ -1384,6 +1384,47 @@ void main() {
     expect(getBackgroundColor(tester, 1), isSameColorAs(CupertinoColors.white));
   });
 
+  // Regression test: https://github.com/flutter/flutter/issues/43414.
+  testWidgets("Quick double tap doesn't break the internal state", (WidgetTester tester) async {
+    const Map<int, Widget> children = <int, Widget>{
+      0: Text('A'),
+      1: Text('B'),
+      2: Text('C'),
+    };
+    int sharedValue = 0;
+
+    await tester.pumpWidget(
+      StatefulBuilder(
+        builder: (BuildContext context, StateSetter setState) {
+          return boilerplate(
+            child: CupertinoSegmentedControl<int>(
+              key: const ValueKey<String>('Segmented Control'),
+              children: children,
+              onValueChanged: (int newValue) {
+                setState(() { sharedValue = newValue; });
+              },
+              groupValue: sharedValue,
+            ),
+          );
+        },
+      ),
+    );
+
+    await tester.tap(find.text('B'));
+    // sharedValue has been updated but widget.groupValue is not.
+    expect(sharedValue, 1);
+
+    // Land the second tap before the widget gets a chance to rebuild.
+    final TestGesture secondTap = await tester.startGesture(tester.getCenter(find.text('B')));
```

Comment:
Is there a reason that the second tap has to be `startGesture` instead of `tap`?   Is this way faster?

Reply:
The sequence of events that leads to this bug is: 

- The first tap finishes, `setState` gets called in the user callback.
- The tap down event of the second tap happens
- The segmented control gets rebuild with the new `groupValue`
- The tap up event of the second tap happens.

The test basically follows these steps so the second tap had to be broken up into a tap down and a tap up event.

Reply:
Got it, thanks for the explanation.